### PR TITLE
Expand product capabilities

### PR DESCRIPTION
**Pull request by fieryprophet**
No issue assigned.

### Description

Adds `create/update/delete` methods to product and product variant endpoints.

### CODE REVIEW

**Review by r-thomson — CHANGES_REQUESTED** [review]

File: bigc/resources/products.py [hunk 1]
```diff
@@ -41,3 +41,22 @@ def get(self, product_id: int, *, include_variants: bool = False, include_custom
         url_parts = url_parts._replace(query=urlencode(query_dict))
 
         return self._v3_client.request('GET', urlunparse(url_parts))
+
+    def create(self, name: str, product_type: str, weight: float, price: float, **kwargs) -> dict:
```
Comment (r-thomson) [inline bigc/resources/products.py:45]:
```suggestion
    def create(self, *, name: str, product_type: str, weight: float, price: float, **kwargs) -> dict:
```

**Review by r-thomson — CHANGES_REQUESTED** [review]
For clarity and consistency, we should disallow passing 'payload' arguments as positional arguments.

File: bigc/resources/products.py [hunk 2]
```diff
@@ -41,3 +41,22 @@ def get(self, product_id: int, *, include_variants: bool = False, include_custom
...
+        payload = {
+            'name': name,
+            'type': product_type,
+            'weight': weight,
+            'price': price,
+            **kwargs,
+        }
+        return self._v3_client.request('POST', '/catalog/products', json=payload)
+
+    def update(self, product_id: int, **kwargs) -> dict:
```
Comment (dhartle4) [inline bigc/resources/products.py:56]: https://github.com/MedShift/bigc/blob/master/bigc/resources/webhooks.py#L28

using a `data` kwarg seems like the established way to pass data into update requests so far.

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "bigc"
-version = "0.1.4"
+version = "0.1.5"
```
Comment (dhartle4) [inline pyproject.toml:3]: I would remove this, Ryan asked that we bump the version numbers in a separate pr.

**Review by dhartle4 — CHANGES_REQUESTED** [review]

File: bigc/resources/products.py (hunk 2 above)
Comment (r-thomson) [reply bigc/resources/products.py]: To be honest, we haven't been particularly consistent.

**Review by dhartle4 — APPROVED** [review]

File: bigc/resources/product_variants.py [hunk 4]
```diff
@@ -14,3 +14,20 @@ def all(self, product_id: int) -> Iterator[dict]:
     def get(self, product_id: int, variant_id: int) -> dict:
         """Get a specific product variant by ID"""
         return self._v3_client.request('GET', f'/catalog/products/{product_id}/variants/{variant_id}')
+
+    def create(self, *, product_id: int, sku: str, option_values: list[dict], **kwargs) -> dict:
```
Comment (r-thomson) [inline bigc/resources/product_variants.py:18]:
```suggestion
    def create(self, product_id: int, *, sku: str, option_values: list[dict], **kwargs) -> dict:
```

**Review by r-thomson — APPROVED** [review]

**Pull request merged**